main.py

```diff
@@ -304,39 +298,54 @@ def get_board(self, access_token_in):
-                                    BytesIO(
-                                        requests.get(
-                                            msg["data"]["name"],
-                                            stream=True,
-                                            proxies=self.GetRandomProxy(),
-                                        ).content
-                                    )
+
+
+        imgs = []       
+        logger.debug("A total of {} canvas sockets opened", len(canvas_sockets))
+        while len(canvas_sockets) > 0:
+            temp = json.loads(ws.recv())
+            logger.debug("Waiting for WebSocket message")
+            if temp["type"] == "data":
+                logger.debug("Received WebSocket data type message")
+                msg = temp["payload"]["data"]["subscribe"]
+                if msg["data"]["__typename"] == "FullFrameMessageData":
+                    logger.debug("Received full frame message")
+                    img_id = int(temp["id"])
+                    logger.debug("Image ID: {}", img_id)
+                    if img_id in canvas_sockets:
+                        logger.debug("Getting image: {}", msg["data"]["name"])
+                        imgs.append(
+                            Image.open(
+                                BytesIO(
+                                    requests.get(
+                                        msg["data"]["name"], stream=True
+                                    ).content
                                 )
                             )
-                            already_added.append(temp["id"])
-                        break
+                        )
+                        canvas_sockets.remove(img_id)
+                        logger.debug("Canvas sockets remaining: {}", len(canvas_sockets))
+
+        for i in range(0, canvas_count-1):
             ws.send(json.dumps({"id": str(2 + i), "type": "stop"}))
 
         ws.close()
 
-        new_img = Image.new("RGB", (1000 * 2, 1000))
-
-        x_offset = 0
-        for img in imgs:
-            new_img.paste(img, (x_offset, 0))
-            x_offset += img.size[0]
-
-        logger.info("Got image.")
-        # new_img.show()
+        #TODO: Multiply by canvas_details["canvasConfigurations"][i]["dx"] and canvas_details["canvasConfigurations"][i]["dy"] instead of hardcoding it
+        new_img_width = int(canvas_details["canvasWidth"]) * 2
+        logger.debug("New image width: {}", new_img_width)
+        new_img_height = int(canvas_details["canvasHeight"])
+        logger.debug("New image height: {}", new_img_height) 
+        
+        new_img = Image.new("RGB", (new_img_width, new_img_height))
+        dx_offset = 0
+        for idx, img in enumerate(imgs):
+            logger.debug("Adding image: {}", img)
+            dx_offset = int(canvas_details["canvasConfigurations"][idx]["dx"])
+            new_img.paste(img, (dx_offset, 0))
+            
+        logger.debug("Saving image to disk as board.png")
```

Comment:
Can we remove that line ?

Reply:
Removed in [603e5fb](https://github.com/rdeepak2002/reddit-place-script-2022/pull/209/commits/603e5fb5a8910338c03cf831e90b4dafd10f05f8)